db.go

```diff
@@ -126,6 +127,9 @@ type DB struct {
 	// changing the autoCompact var.
 	autoCompactMtx sync.Mutex
 	autoCompact    bool
+
+	// Cancel a running compaction when a shutdown is initiated.
+	compactCnl func()
```

Comment:
What about moving to something all use in Golang:

```suggestion
	compactCancel context.CancelFunc
```

Reply:
updated.